### PR TITLE
Api 9 postgres schema

### PR DESCRIPTION
**Pull request by dethe**
NOT READY TO MERGE

This is the work-in progress for creating an SQL schema for the existing data, and will likely require some iteration.

### CODE REVIEW

File: tables.sql [hunk 1]
```diff
@@ -0,0 +1,187 @@
+/* CASES */
+
+CREATE TABLE cases (
+    id INTEGER PRIMARY KEY SERIAL,
+    body_en TEXT, /* are we going to store every language in this table? */
```
Comment (davidascher) [inline tables.sql]: Title and body are the two localizable fields.  My assumption is we'll want to store those in different tables, but we should discuss.

File: tables.sql [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    title_en TEXT,
+    completeness COMPLETE_STATE NOT NULL,
```
Comment (davidascher) [inline tables.sql]: this field should go.

File: tables.sql [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+CREATE TABLE communication_modes (
+    -- id INTEGER PRIMARY KEY SERIAL,
+    mode TEXT NOT NULL,
+    case_id INTEGER NOT NULL REFERENCES cases
+);
+
+CREATE TYPE complete_state as ENUM ('stub', 'partial', 'complete');
+CREATE TYPE virtualness as ENUM ('face-to-face', 'online', 'both');
+CREATE TYPE geo_scope as ENUM ('International', 'Local (e.g. Neighbourhood, City/Town, Metropolitan Area)', 'National', 'Regional (e.g. State, Province, Autonomous Region)');
+CREATE TYPE voting_type as ENUM ('None', 'Consensus', 'Super-Majoritarian Voting (i.e. threshold more than 50%+1)', 'Majoritarian Voting (i.e. 50%+1)', 'Preferential Voting (i.e. ranking preferences)');
```
Comment (davidascher) [inline tables.sql]: I would imagine we'll want to encode these things with key names that are used to look up the natural language label in a translation table, no?  Would simplify this bit to for example not need spaces, parens, etc in the enums.

File: tables.sql [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+CREATE TABLE content_languages (
```
Comment (davidascher) [inline tables.sql]: I don't understand this table.

File: tables.sql (hunk 1 above)
Comment (dethe) [inline tables.sql]: We're not going to localize the rest? That seems odd, but maybe OK for a first cut.

Getting the query  right for finding localized content, but falling back to non-local when localized is not available will be interesting.

File: tables.sql (hunk 3 above)
Comment (dethe) [inline tables.sql]: Yeah, I was putting in what I was getting from the old data, but using lookup keys would be preferable.

If we get rid of the enums altogether and store the default values outside the DB (or in a separate, localizable, table), is that for lists with "other" options we can store the "other" responses in the same table.

File: tables.sql (hunk 4 above)
Comment (dethe) [inline tables.sql]: Each data type (case, method, organization), has a field in the current data for content languages as a list. This was my attempt to preserve it.

Comment (dethe) [inline tables.sql]: It was a value used 99% of the time, for what that's worth.

File: migrations/cases.sql [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+CREATE TYPE virtualness as ENUM ('facetoface', 'online', 'both');
+CREATE TYPE geo_scope as ENUM ('international', 'local', 'national', 'regional');
+CREATE TYPE voting_type as ENUM ('none', 'consensus', 'supermajority', 'majority', 'preferential');
+CREATE TYPE filetype as ENUM ('file', 'image');
+
+
+CREATE TABLE cases (
+    id INTEGER PRIMARY KEY SERIAL,
+    original_language TEXT, /* two-letter IANA language subtag */
+    content_country TEXT,
```
Comment (davidascher) [inline migrations/cases.sql]: is this not implicit in the geolocation?

File: migrations/cases.sql [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    case_id INTEGER NOT NULL REFERENCES cases
+)
+
+/* all methods referenced by a case */
+CREATE TABLE case__methods (
+    method_id INTEGER NOT NULL REFERENCES methods
+    case_id INTEGER NOT NULL REFERENCES cases,
+);
+
+/* Keep track of each day (and time?) someone edits */
```
Comment (davidascher) [inline migrations/cases.sql]: yes, time, for e.g. edit wars etc.

File: migrations/cases.sql [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+    case_id INTEGER NOT NULL REFERENCES cases,
+);
+
+CREATE TABLE case__communication_modes (
+    mode TEXT NOT NULL,
+    case_id INTEGER NOT NULL REFERENCES cases
+);
+
+CREATE TABLE case__decision_methods (
+    -- id INTEGER PRIMARY KEY SERIAL,
```
Comment (davidascher) [inline migrations/cases.sql]: stray comment?

File: migrations/cases.sql [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+CREATE TABLE case__attachments (
+    url TEXT NOT NULL,
+    title TEXT,
+    type filetype NOT NULL,
+    size INTEGER,
+    is_lead BOOLEAN,
+    case_id INTEGER NOT NULL REFERENCES cases
+)
+
+CREATE TABLE geolocation (
```
Comment (davidascher) [inline migrations/cases.sql]: I think we may want to encode something like the resolution of a geopoint -- address, municipal, state,etc. See the `types` in the geocoding API docs: https://developers.google.com/maps/documentation/geocoding/intro#reverse-response

File: migrations/methods.sql [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+CREATE TABLE methods (
+    id INTEGER PRIMARY KEY SERIAL,
+    original_language TEXT, /* two-letter IANA language subtag */
+    facilitated BOOLEAN,
+    geographical_scope TEXT,
+    issue_interdependency TEXT,
+    issue_polarization TEXT,
+    issue_technical_complexity TEXT,
+    -- path TEXT NOT NULL, /* path to this method */
```
Comment (davidascher) [inline migrations/methods.sql]: We should talk about URL slugs for all of these things, or whether we are OK w/ putting IDs in URLs.

**Pull request closed without merging**